File: programmers/DFS-BFS/타겟넘버.py
answer=0
def dfs(x, cnt,dp,y):
    global answer
    if cnt == len(dp):
        if x == y:
            answer += 1
        return
    dfs(x+dp[cnt], cnt+1,dp,y)
    dfs(x-dp[cnt], cnt+1,dp,y)
    return

def solution(numbers, target):
    dfs(0, 0,numbers,target)
    return answer

File: javascript/programmers/DFS_BFS/타겟넘버.js
function solution(numbers, target) {
    var answer = 0;
    function check(cnt, num) {
        if (cnt == numbers.length) {
            if (num == target) {
                answer++;
            }
            return;
        }
        check(cnt + 1, num - numbers[cnt]);
        check(cnt + 1, num + numbers[cnt]);
    }
    check(0, 0);
    return answer;
}
